### ISSUE TITLE
Nuxt Module: Unable to import dependencies that contain aliased imports

### ISSUE THREAD

**Issue by dombavetta** [nuxt/nuxt.js#8941]
### Versions

- nuxt: v2.15.1
- node: v12.14.0

### Reproduction

https://codesandbox.io/s/nuxt-module-require-with-alias-9o73j?file=/modules/issue-module.js

<details open>
<summary>Additional Details</summary>
<br>

Error produced -
(may need to restart sandbox to see the error)

```
Cannot find module '~/app/base-unit'                          
Require stack:
- /sandbox/app/foo-unit.js

  Require stack:
  - app/foo-unit.js
  at Function.Module._resolveFilename (internal/modules/cjs/loader.js:880:15)
  at Function.resolve (internal/modules/cjs/helpers.js:94:19)
  at l (node_modules/jiti/dist/jiti.js:1:8109)
  at p (node_modules/jiti/dist/jiti.js:1:8505)
  at app/foo-unit.js:1:179
  at p (node_modules/jiti/dist/jiti.js:1:9442)
  at ModuleContainer._default (modules/issue-module.js:8:22)
  at ModuleContainer.addModule (node_modules/@nuxt/core/dist/core.js:239:34)
  at node_modules/@nuxt/utils/dist/utils.js:638:43
  at async ModuleContainer.ready (node_modules/@nuxt/core/dist/core.js:51:7)
```

</details>

### Steps to reproduce
- Create two files, one of which imports the other using one of Nuxt's preconfigured webpack resolve aliases `~/` or `@/`
- Create Nuxt module which uses `require('path to module w/ alias import')` to import the js module
- Register module in `nuxt.config.js`
- Start development server
- See terminal output

### What is Expected?
The aliased path is successfully resolved and the dependency module is retrieved. 

### What is actually happening?
The js module fails to be retrieved producing a fatal error

```

Cannot find module '~/app/base-unit'                          
Require stack:
- /sandbox/app/foo-unit.js

```

**Comment by danielroe** [comment]
@dombavetta modules run 'outside' of the webpack context and if they import files they don't benefit from these aliases. However, you can benefit from the built-in Nuxt resolver, e.g.:
```js
const unitModule = this.nuxt.resolver.resolveAlias(unitPath);
```

**Comment by dombavetta** [comment]
@danielroe  `resolveAlias` is definitely a neat utility that I was not familiar with, will be keeping that in my back pocket! For my use case, I have my app broken up based on defined "business units". I have folders 

```
app/
  - unit1
  - unit2
  - unit3
```

Each unit folder contains constructs like components, store, services, etc.. isolated to the associated business unit. Each unit also has an index file that exports a few things to help initialize the unit. Within a module I was trying to import these index files to help automate some app bootstrapping but was blocked by the alias within issue.

**Comment by pi0** [comment]
@dombavetta You can also take a look at [nuxt-extend](https://github.com/nuxt-community/nuxt-extend) for multi app support. It gives each unit/app a unique alias to make combustion easier.

**Comment by dombavetta** [comment]
@pi0  While that util may actually be useful within my application, it kind of services a different problem. Let me dive a little deeper into a specific of my implementation. While maybe not being recommended, part of the application relies on constructor names being static. During unit initialization I have middleware that performs an action like so `useUnit(MyBusinessUnit)` where `MyBusinessUnit` is a `class`. `useUnit` essentially constructs an instance of the class injecting its dependencies and then registers it within a collection so that it can just be looked up rather than reinitialized on subsequent uses. This relates to Nuxt by me needing to configure terser to not mangle the class names. As part of the terser options `keep_classnames` and `keep_fnames` you can provide a regex to basically whitelist names to not be mangled during the build. One step in my nuxt module attempts to resolve all the "units" (index.ts) under `app/units/{unit}` and then create a regex based on all the imported class names. This is where I face the issue described above. For my use case, there are certainly ways around this with a little bit of refactoring but I was more so curious to see if there was an actual solution to this as I feel that this cant be the only case where one would want to import a js module with aliased imports within a nuxt module.

Side note: I really do appreciate the efforts made by the team to resolve my specific use case - between you, Sébastien and @danielroe I've learned about handful or neat utilities that I will surely find useful in future applications!

**Comment by shtse8** [comment]
how can I import dependencies in nuxt module that contain aliased imports?

**Comment by ozgurg** [comment]
I have similar issue.

My `nuxt.config.js` like this:
```javascript
import myDynamicRoutes from "@data/my-dynamic-routes.js";

export default {
  ...
  generate: {
    routes: myDynamicRoutes
  }
  ...
}
```

`data/my-dynamic-routes.js`:
```javascript
import anotherFile from "@utils/another-file.js";

const routes = [
  "/route-1",
  "/route-2",
  ...anotherFile
];

export default routes;
```

`utils/another-file.js`:
```javascript
const variable = ["/a", "/b", "/c"];

export default variable;
```

When I run "nuxt generate", it gives me:
`Cannot find module '@data/my-dynamic-routes.js'`

I tried @danielroe's way, it fixed the issue by creating another issue. @data/my-dynamic-routes.js successfully imported with "resolveAlias" way but @utils/another-file.js file inside of it gives me look a like error:
`Cannot find module '@utils/another-file.js'`
